app/services/email_verification_code_service.rb

```diff
@@ -0,0 +1,69 @@
+class EmailVerificationCodeService
```

Comment:
This is a port from ClientLoginsService to create distinct services for Email and TextMessage codes. It allows for passing a validation_type, which alters the message and the logic for how it checks to see whether you are "eligible" for a code.

There is a lot of commonality between the text and email logic, but i decided to leave the refactor in this state due to time constraints and to prevent divergence from the original implementation too much.